### PR TITLE
SchedulesDirect fix

### LINKED ISSUE #8502: [Issue]: Unable to get SchedulesDirect Data
### Please describe your bug

When I run TV Guide update my system throws a much of HTTP exceptions and no data is collected from SchedulesDirect.

I am able to see the Lineup options, but am unable to get the Schedule data.

I have updated and restarted the server, as well as deleted and re-created all of the Live TV settings

### Jellyfin Version

10.8.0

### if other:

_No response_

### Environment

```markdown
- OS: Docker
- Virtualization: Docker
- Clients: All
- Browser: Chrome
- FFmpeg Version:
- Playback Method:
- Hardware Acceleration:
- Plugins:
- Reverse Proxy:
- Base URL:
- Networking: Host
- Storage: Local
```

### Jellyfin logs

```shell
15:46:53] [INF] [26] Emby.Server.Implementations.LiveTv.Listings.SchedulesDirect: Authenticated with Schedules Direct token: REDACTED
[15:46:53] [ERR] [26] Emby.Server.Implementations.LiveTv.LiveTvManager: Error getting programs for channel MYST
System.InvalidOperationException: The request message was already sent. Cannot send the same request message multiple times.
   at System.Net.Http.HttpClient.CheckRequestMessage(HttpRequestMessage request)
   at System.Net.Http.HttpClient.CheckRequestBeforeSend(HttpRequestMessage request)
   at Emby.Server.Implementations.LiveTv.Listings.SchedulesDirect.Send(HttpRequestMessage options, Boolean enableRetry, ListingsProviderInfo providerInfo, CancellationToken cancellationToken, HttpCompletionOption completionOption)
   at Emby.Server.Implementations.LiveTv.Listings.SchedulesDirect.Send(HttpRequestMessage options, Boolean enableRetry, ListingsProviderInfo providerInfo, CancellationToken cancellationToken, HttpCompletionOption completionOption)
   at Emby.Server.Implementations.LiveTv.Listings.SchedulesDirect.GetChannels(ListingsProviderInfo info, CancellationToken cancellationToken)
   at Emby.Server.Implementations.LiveTv.EmbyTV.EmbyTV.GetEpgChannels(IListingsProvider provider, ListingsProviderInfo info, Boolean enableCache, CancellationToken cancellationToken)
   at Emby.Server.Implementations.LiveTv.EmbyTV.EmbyTV.GetEpgChannelFromTunerChannel(IListingsProvider provider, ListingsProviderInfo info, ChannelInfo tunerChannel, CancellationToken cancellationToken)
   at Emby.Server.Implementations.LiveTv.EmbyTV.EmbyTV.GetProgramsAsync(String channelId, DateTime startDateUtc, DateTime endDateUtc, CancellationToken cancellationToken)
   at Emby.Server.Implementations.LiveTv.LiveTvManager.RefreshChannelsInternal(ILiveTvService service, IProgress`1 progress, CancellationToken cancellationToken)
[15:46:53] [INF] [26] Emby.Server.Implementations.LiveTv.Listings.SchedulesDirect: Authenticated with Schedules Direct token: REDACTED
[15:46:53] [ERR] [26] Emby.Server.Implementations.LiveTv.EmbyTV.EmbyTV: Error adding metadata
System.InvalidOperationException: The request message was already sent. Cannot send the same request message multiple times.
   at System.Net.Http.HttpClient.CheckRequestMessage(HttpRequestMessage request)
   at System.Net.Http.HttpClient.CheckRequestBeforeSend(HttpRequestMessage request)
   at Emby.Server.Implementations.LiveTv.Listings.SchedulesDirect.Send(HttpRequestMessage options, Boolean enableRetry, ListingsProviderInfo providerInfo, CancellationToken cancellationToken, HttpCompletionOption completionOption)
   at Emby.Server.Implementations.LiveTv.Listings.SchedulesDirect.Send(HttpRequestMessage options, Boolean enableRetry, ListingsProviderInfo providerInfo, CancellationToken cancellationToken, HttpCompletionOption completionOption)
   at Emby.Server.Implementations.LiveTv.Listings.SchedulesDirect.GetChannels(ListingsProviderInfo info, CancellationToken cancellationToken)
   at Emby.Server.Implementations.LiveTv.EmbyTV.EmbyTV.GetEpgChannels(IListingsProvider provider, ListingsProviderInfo info, Boolean enableCache, CancellationToken cancellationToken)
   at Emby.Server.Implementations.LiveTv.EmbyTV.EmbyTV.AddMetadata(IListingsProvider provider, ListingsProviderInfo info, IEnumerable`1 tunerChannels, Boolean enableCache, CancellationToken cancellationToken)
   at Emby.Server.Implementations.LiveTv.EmbyTV.EmbyTV.GetChannelsAsync(Boolean enableCache, CancellationToken cancellationToken)
[15:46:53] [INF] [26] Emby.Server.Implementations.LiveTv.Listings.SchedulesDirect: Authenticated with Schedules Direct token: REDACTED
[15:46:53] [ERR] [26] Emby.Server.Implementations.LiveTv.LiveTvManager: Error getting programs for channel DABL
System.InvalidOperationException: The request message was already sent. Cannot send the same request message multiple times.
   at System.Net.Http.HttpClient.CheckRequestMessage(HttpRequestMessage request)
   at System.Net.Http.HttpClient.CheckRequestBeforeSend(HttpRequestMessage request)
   at Emby.Server.Implementations.LiveTv.Listings.SchedulesDirect.Send(HttpRequestMessage options, Boolean enableRetry, ListingsProviderInfo providerInfo, CancellationToken cancellationToken, HttpCompletionOption completionOption)
   at Emby.Server.Implementations.LiveTv.Listings.SchedulesDirect.Send(HttpRequestMessage options, Boolean enableRetry, ListingsProviderInfo providerInfo, CancellationToken cancellationToken, HttpCompletionOption completionOption)
   at Emby.Server.Implementations.LiveTv.Listings.SchedulesDirect.GetChannels(ListingsProviderInfo info, CancellationToken cancellationToken)
   at Emby.Server.Implementations.LiveTv.EmbyTV.EmbyTV.GetEpgChannels(IListingsProvider provider, ListingsProviderInfo info, Boolean enableCache, CancellationToken cancellationToken)
   at Emby.Server.Implementations.LiveTv.EmbyTV.EmbyTV.GetEpgChannelFromTunerChannel(IListingsProvider provider, ListingsProviderInfo info, ChannelInfo tunerChannel, CancellationToken cancellationToken)
   at Emby.Server.Implementations.LiveTv.EmbyTV.EmbyTV.GetProgramsAsync(String channelId, DateTime startDateUtc, DateTime endDateUtc, CancellationToken cancellationToken)
   at Emby.Server.Implementations.LiveTv.LiveTvManager.RefreshChannelsInternal(ILiveTvService service, IProgress`1 progress, CancellationToken cancellationToken)
[15:46:53] [INF] [26] Emby.Server.Implementations.ScheduledTasks.TaskManager: Refresh Guide Completed after 0 minute(s) and 17 seconds
```

### FFmpeg logs

_No response_

### Please attach any browser or client logs here

_No response_

### Please attach any screenshots here

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by barronpm**
**Changes**
- Maybe fixes #8502 
- Fixes some analyzer warnings
- Depends on #10838

### CODE REVIEW

**Review by MarkCiliaVincenti — APPROVED** [review]

**Comment by mega-mac-slice** [comment]
Would it be possible to log/include the response body/payload on >=400? It seems like we're just logging `Bad Request` here and some more useful context might be in the SD response to help individuals troubleshoot.

**Comment by barronpm** [comment]
> Would it be possible to log/include the response body/payload and request url on >=400? It seems like we're just logging `Bad Request` here and some more useful context might be in the SD response to help individuals troubleshoot.

I've added a log statement that should help, can you test to double-check it has what you're looking for?

**Comment by mega-mac-slice** [comment]
Something like this - otherwise we get the class name representation:
```
-                _logger.LogError("Request to {Url} failed with response {Response}", message.RequestUri, response.Content);
+                _logger.LogError("Request to {Url} failed with response {1}", message.RequestUri, await response.Content.ReadAsStringAsync().ConfigureAwait(false));
```
(Entirely possibly there's a better way to do what I'm doing.)

```
[16:57:05] [ERR] [18] Jellyfin.LiveTv.Listings.SchedulesDirect: Request to https://json.schedulesdirect.org/20141201/lineups/USA-OTA-77879 failed with response {"response":"LINEUP_NOT_FOUND","code":2101,"serverID":"20141201.web","message":"Lineup not in account. Add lineup to account before requesting mapping.","datetime":"2024-01-24T22:57:05Z"}
[16:57:08] [ERR] [18] Jellyfin.Api.Middleware.ExceptionMiddleware: Error processing request. URL GET /LiveTv/ChannelMappingOptions.
System.Net.Http.HttpRequestException: Request failed: Bad Request
   at Jellyfin.LiveTv.Listings.SchedulesDirect.Request[T](HttpRequestMessage message, Boolean enableRetry, ListingsProviderInfo providerInfo, CancellationToken cancellationToken, HttpCompletionOption completionOption) in /home/me/dev/src/jellyfin/src/Jellyfin.LiveTv/Listings/SchedulesDirect.cs:line 613
   at Jellyfin.LiveTv.Listings.SchedulesDirect.Request[T](HttpRequestMessage message, Boolean enableRetry, ListingsProviderInfo providerInfo, CancellationToken cancellationToken, HttpCompletionOption completionOption) in /home/me/dev/src/jellyfin/src/Jellyfin.LiveTv/Listings/SchedulesDirect.cs:line 626
   at Jellyfin.LiveTv.Listings.SchedulesDirect.GetChannels(ListingsProviderInfo info, CancellationToken cancellationToken) in /home/me/dev/src/jellyfin/src/Jellyfin.LiveTv/Listings/SchedulesDirect.cs:line 735
   at Jellyfin.Api.Controllers.LiveTvController.GetChannelMappingOptions(String providerId) in /home/me/dev/src/jellyfin/Jellyfin.Api/Controllers/LiveTvController.cs:line 1093
   at lambda_method1143(Closure, Object)
```

**Comment by mega-mac-slice** [comment]
https://github.com/jellyfin/jellyfin/issues/8502#issuecomment-1909109639 - found another spot we might want to log errors - https://github.com/barronpm/jellyfin/blob/bfab1372a5747db44903889f8cb23a47ac68c858/src/Jellyfin.LiveTv/Listings/SchedulesDirect.cs#L664

**Comment by mega-mac-slice** [comment]
Running through some final testing for your latest changes:

## Existing Lineup X

Had an existing lineup on my dev instance from yesterday - honestly wasn't sure what state it was in - but is apparently bad:
Screenshot from 2024-01-25 16-41-30

I click on "Map Channels" and get a broken state:
Screenshot from 2024-01-25 16-42-04

I check the server logs and find useful information to help me diagnose the issue further:
```
[16:37:35] [ERR] [39] Jellyfin.LiveTv.Listings.SchedulesDirect: Request to https://json.schedulesdirect.org/20141201/lineups/USA-OTA-77385 failed with response {"response":"LINEUP_NOT_FOUND","code":2101,"serverID":"20141201.web","message":"Lineup not in account. Add lineup to account before requesting mapping.","datetime":"2024-01-25T22:37:35Z"}
[16:37:37] [ERR] [39] Jellyfin.Api.Middleware.ExceptionMiddleware: Error processing request. URL GET /LiveTv/ChannelMappingOptions.
System.Net.Http.HttpRequestException: Request failed: Bad Request
   at Jellyfin.LiveTv.Listings.SchedulesDirect.Request[T](HttpRequestMessage message, Boolean enableRetry, ListingsProviderInfo providerInfo, CancellationToken cancellationToken, HttpCompletionOption completionOption) in /home/me/dev/src/jellyfin/src/Jellyfin.LiveTv/Listings/SchedulesDirect.cs:line 616
   at Jellyfin.LiveTv.Listings.SchedulesDirect.Request[T](HttpRequestMessage message, Boolean enableRetry, ListingsProviderInfo providerInfo, CancellationToken cancellationToken, HttpCompletionOption completionOption) in /home/me/dev/src/jellyfin/src/Jellyfin.LiveTv/Listings/SchedulesDirect.cs:line 629
   at Jellyfin.LiveTv.Listings.SchedulesDirect.GetChannels(ListingsProviderInfo info, CancellationToken cancellationToken) in /home/me/dev/src/jellyfin/src/Jellyfin.LiveTv/Listings/SchedulesDirect.cs:line 749
   at Jellyfin.Api.Controllers.LiveTvController.GetChannelMappingOptions(String providerId) in /home/me/dev/src/jellyfin/Jellyfin.Api/Controllers/LiveTvController.cs:line 1093
   at lambda_method1108(Closure, Object)
```

## Update Existing Lineup X

I assume I have available lineup options in my SD account - so I'm going to try re-saving it - changing nothing - I assume this is going to make another attempt at adding the lineup and be successful:
Screenshot from 2024-01-25 16-48-23

This appears to save correctly and kick off the refresh task - truncated logs:
```
[16:48:40] [INF] [47] Jellyfin.LiveTv.Listings.SchedulesDirect: Headends on account 
[16:48:40] [INF] [47] Jellyfin.LiveTv.Listings.SchedulesDirect: Adding new lineup USA-OTA-77385
[16:48:43] [INF] [48] Jellyfin.LiveTv.EmbyTV.EmbyTV: Discovered tuner device HD Homerun at http://192.168.1.197
[16:48:43] [INF] [48] Jellyfin.LiveTv.EmbyTV.EmbyTV: Discovered tuner device HD Homerun at http://192.168.1.197
[16:48:44] [INF] [44] Jellyfin.LiveTv.Listings.SchedulesDirect: Found 123 channels on the lineup on ScheduleDirect
[16:48:44] [INF] [44] Jellyfin.LiveTv.Listings.SchedulesDirect: Mapping Stations to Channel
[16:48:44] [INF] [44] Jellyfin.LiveTv.EmbyTV.EmbyTV: Found epg channel in Schedules Direct USA-OTA-77385 KPRC-DT 21250
[16:48:44] [INF] [44] Jellyfin.LiveTv.EmbyTV.EmbyTV: Found epg channel in Schedules Direct USA-OTA-77385 KPRC-DT2 40465
...
[16:48:44] [INF] [44] Jellyfin.LiveTv.EmbyTV.EmbyTV: Found epg channel in Schedules Direct USA-OTA-77385 KFTH-DT4 105775
[16:48:44] [INF] [44] Jellyfin.LiveTv.EmbyTV.EmbyTV: Found epg channel in Schedules Direct USA-OTA-77385 KFTH-DT5 111329
[16:48:44] [INF] [44] Jellyfin.LiveTv.LiveTvManager: Refreshing guide with 7 days of guide data
[16:48:44] [INF] [44] Jellyfin.LiveTv.Listings.SchedulesDirect: Channel Station ID is: 42574
...
[16:48:55] [INF] [48] Jellyfin.LiveTv.Listings.SchedulesDirect: Channel Station ID is: 95562
[16:48:56] [INF] [48] Emby.Server.Implementations.ScheduledTasks.TaskManager: Refresh Guide Completed after 0 minute(s) and 16 seconds
```

## Creating New Lineup When At SD Lineup Capacity

I assumed I had room on SD account to make a good test case - but was incorrect:
Screenshot from 2024-01-25 16-55-02

I check the server logs and find useful information to help me diagnose the issue further:
```
[16:55:34] [INF] [42] Jellyfin.LiveTv.Listings.SchedulesDirect: Adding new lineup USA-OTA-78719
[16:55:34] [ERR] [42] Jellyfin.LiveTv.Listings.SchedulesDirect: Error adding lineup to account: {"response":"MAX_LINEUPS","code":4101,"serverID":"20141201.web","message":"Exceeded number of lineups for this account.","datetime":"2024-01-25T22:55:34Z"}
```

## Creating New Lineup 

Use SD API to remove a lineup from my account - so I'm not at my limit.

Create the following lineup:
Screenshot from 2024-01-25 17-13-52

```
[17:14:36] [INF] [40] Jellyfin.LiveTv.Listings.SchedulesDirect: Adding new lineup USA-OTA-78940
[17:14:39] [INF] [29] Jellyfin.LiveTv.EmbyTV.EmbyTV: Discovered tuner device HD Homerun at http://192.168.1.197
```
There's a lot of logs from a schedule refresh attempting to be ran when then the form is being filled out on an empty listing - but I'm pretty sure I saw this before your changes so believe it's unrelated. There's also a bunch of noise from my existing bad lineups trying to run and failing. But overall this is good.

Too better confirm it's in a good state I check the mappings to assure the endpoint is now resolving:
Screenshot from 2024-01-25 17-19-34


## TLDR

Overall I'm happy with these changes - it better surfaces the underlying issues so that user's can diagnose further. Let me know if there's any further testing required/desired - and thanks again for adding the response logs.

File: src/Jellyfin.LiveTv/Listings/SchedulesDirect.cs [hunk 1]
```diff
@@ -597,36 +592,41 @@ private async Task<string> GetToken(ListingsProviderInfo info, CancellationToken
...
+        private async Task<T> Request<T>(
+            HttpRequestMessage message,
             bool enableRetry,
             ListingsProviderInfo providerInfo,
             CancellationToken cancellationToken,
             HttpCompletionOption completionOption = HttpCompletionOption.ResponseContentRead)
         {
-            var response = await _httpClientFactory.CreateClient(NamedClient.Default)
-                .SendAsync(options, completionOption, cancellationToken).ConfigureAwait(false);
+            using var client = _httpClientFactory.CreateClient(NamedClient.Default);
```
Comment (crobibero) [inline src/Jellyfin.LiveTv/Listings/SchedulesDirect.cs:602]: Shouldn't dispose the HttpClient

File: src/Jellyfin.LiveTv/Listings/SchedulesDirect.cs [hunk 2]
```diff
@@ -662,11 +660,22 @@ private async Task AddLineupToAccount(ListingsProviderInfo info, CancellationTok
...
-            _logger.LogInformation("Adding new LineUp ");
+            _logger.LogInformation("Adding new lineup {Id}", info.ListingsId);
 
-            using var options = new HttpRequestMessage(HttpMethod.Put, ApiUrl + "/lineups/" + info.ListingsId);
-            options.Headers.TryAddWithoutValidation("token", token);
-            using var response = await _httpClientFactory.CreateClient(NamedClient.Default).SendAsync(options, HttpCompletionOption.ResponseHeadersRead, cancellationToken).ConfigureAwait(false);
+            using var message = new HttpRequestMessage(HttpMethod.Put, ApiUrl + "/lineups/" + info.ListingsId);
+            message.Headers.TryAddWithoutValidation("token", token);
+
+            using var client = _httpClientFactory.CreateClient(NamedClient.Default);
```
Comment (crobibero) [inline src/Jellyfin.LiveTv/Listings/SchedulesDirect.cs:668]: Shouldn't dispose the HttpClient

**Review by crobibero — APPROVED** [review]

**Review by Bond-009 — APPROVED** [review]
